inst/templates/ad_admh.R

```diff
@@ -0,0 +1,173 @@
+    max_dates = vars(DTHDT, EOSDT)
+  ) %>%
+  # Derive analysis start relative day and analysis end relative day
+  derive_vars_dy(
+    reference_date = TRTSDT,
+    source_vars = vars(ASTDT, AENDT)
+  ) %>%
+  # Derive analysis date of medical history collection - ADT (company specific variable derivation)
+  derive_vars_dt(
+    dtc = MHDTC,
+    new_vars_prefix = "A"
+  ) %>%
+  # Derive analysis relative day - ADY (company specific variable derivation)
+  derive_vars_dy(
+    reference_date = TRTSDT,
+    source_vars = vars(ADT)
+  ) %>%
+  # Derive query variables
+  derive_vars_query(queries_mh) %>%
+  # Assign the AHIST (company specific variable derivation)
+  mutate(AHIST = case_when(
+    MHENRF == "BEFORE" ~ "Past",
+    MHENRF %in% c("DURING", "AFTER") ~ "Current",
+    MHSTAT == "Not Done" ~ "Not Assessed"
+  )) %>%
+  # Derive occurrence flags
+  derive_var_extreme_flag(
+    by_vars = vars(USUBJID),
+    order = vars(ASTDT, MHSEQ),
+    new_var = AOCCFL,
+    mode = "first"
+  ) %>%
+  derive_var_extreme_flag(
+    by_vars = vars(USUBJID, MHBODSYS),
+    order = vars(USUBJID, MHBODSYS, MHCAT, MHDECOD, MHTERM, ASTDT, MHSEQ),
+    new_var = AOCCSFL,
+    mode = "first"
+  ) %>%
+  derive_var_extreme_flag(
+    by_vars = vars(USUBJID, MHDECOD),
+    order = vars(USUBJID, MHBODSYS, MHCAT, MHDECOD, MHTERM, ASTDT, MHSEQ),
+    new_var = AOCCPFL,
+    mode = "first"
+  ) %>%
+  # (company specific occurrence flag variables derivation)
+  derive_var_extreme_flag(
+    by_vars = vars(USUBJID),
+    order = vars(USUBJID, AHIST, ASTDT, MHSEQ),
+    new_var = AOCPFL,
+    mode = "first"
+  ) %>%
+  derive_var_extreme_flag(
+    by_vars = vars(USUBJID, MHBODSYS),
+    order = vars(USUBJID, AHIST, MHBODSYS, MHCAT, ASTDT, MHSEQ),
+    new_var = AOCPSFL,
+    mode = "first"
+  ) %>%
+  derive_var_extreme_flag(
+    by_vars = vars(USUBJID, MHDECOD),
+    order = vars(USUBJID, AHIST, MHBODSYS, MHCAT, MHDECOD, MHTERM, ASTDT, MHSEQ),
+    new_var = AOCPPFL,
+    mode = "first"
+  ) %>%
+  # Derive analysis flag (company specific variable derivation)
+  mutate(ANL01FL = ifelse(MHOCCUR != "N", "Y", NA_character_)) %>%
+  # Assign TRTA, TRTP (company specific variables derivation)
+  mutate(
+    TRTP = TRT01P,
+    TRTA = TRT01A
+  ) %>%
+  # Assign APHASE and APHASEN Variable (company specific variable derivation)
+  mutate(
+    APHASE = case_when(
+      ADT < TRTSDT ~ "Screening",
+      ADT > TRTEDT ~ "Post-Treatment",
+      ADT <= TRTSDT & ADT >= TRTEDT ~ "On-Treatment"
+    ),
+    APHASEN = case_when(
+      ADT < TRTSDT ~ 1,
+      ADT > TRTEDT ~ 2,
+      ADT <= TRTSDT & ADT >= TRTEDT ~ 3
+    )
+  )
+
+# Derive MHTERMN (company specific variable derivation)
+admh <- admh %>%
+  filter(MHPRESP == "Y") %>%
+  derive_vars_merged(
+    dataset_add = mhtermn_lookup,
+    new_vars = vars(MHTERMN),
+    by_vars = vars(MHTERM)
+  )
+
+admh <- admh %>%
+  filter(is.na(MHPRESP) | MHPRESP != "Y") %>%
+  mutate(MHTERMN = NA) %>%
+  bind_rows(admh)
+
```

Comment:
The above two steps are subsetting the admh dataset (admiral_mh has 1818 records, and admh only has 254 in the end). Do we need to filter the dataset? I thought if the mhterm_lookup table doesn't contain the `MHTERM` a `MHTERMN` will be `NA` anyways. If we need to filter, could we leverage the `restrict_derivation()` instead of directly subsetting the dataset?

Reply:
Definitely make use of `restrict_derivation()` if filtering is needed.

Reply:
Thanks for reviewing @yangc71. Filtering the dataset is required as MHTERMN will be assigned only when MHPRESP='Y'. I've updated the code by using restrict_derivation() as per your suggestion.